lib/resources/os.rb

```diff
@@ -54,5 +54,58 @@ def params
     def to_s
       'Operating System Detection'
     end
+
+    # Windows release numbers from https://en.wikipedia.org/wiki/Ver_(command)
+    # 6.0.6001 6.0.6002
+    def windows_server_2008?
+      return true if release =~ /6.0.600/
+      false
+    end
+
+    # 6.1.7600 6.1.7600.16385 6.1.7601
+    def windows_7?
+      return true if release =~ /6.1.760/
+      false
+    end
+
+    # 6.1.7600 6.1.7600.16385 6.1.7601
+    def windows_server_2008_r2?
+      windows_7?
+    end
+
+    # 6.2.9200 6.2.9200.16384
+    def windows_8?
+      return true if release =~ /6.2.9200/
+      false
+    end
+
+    # 6.2.9200
+    def windows_server_2012?
+      return true if release =~ /6.2.9200/
+      false
+    end
+
+    # 6.3.9600
+    def windows_8_1?
+      return true if release =~ /6.3.9600/
+      false
+    end
+
+    # 6.3.9600
+    def windows_server_2012_r2?
+      windows_8_1?
```

Comment:
This is not a safe way to implement this. I would rely on https://github.com/chef/train/blob/master/lib/train/extras/os_detect_windows.rb#L51

Reply:
I was using the fact that windows_8_1 and 2012_r2 are the same to reduce the code footprint. I can revert it to duplicated code if that's what you mean.

Reply:
I should probably also add a windows? check to all of these too.